ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java

```diff
@@ -62,6 +64,7 @@ public class BinPackedPersistentQueryMetadataImpl implements PersistentQueryMeta
   private static final Logger LOG = LoggerFactory
       .getLogger(BinPackedPersistentQueryMetadataImpl.class);
 
+  private final AtomicBoolean isPaused = new AtomicBoolean(false);
```

Comment:
Are we actually expecting concurrent access to this value? ksql processing statements from the command topic should be single-threaded, so I'd expect at least updates to this value to only be from a single thread. It's possible that there are other threads which might try to read the value concurrently. (Curious what your thinking here was.)

Reply:
Ah!  I didn't know for sure that this class is used in a single-threaded manner.  So I assumed that an AtomicBoolean was better/more sensible.  

If there's multi-threaded access, I can see that things may be a mess for other things in class.

I'll switch from an AtomicBoolean to a regular one.